lib/request_code.dart

```diff
@@ -25,7 +25,9 @@ class RequestCode {
     _code = null;
 
     final urlParams = _constructUrlParams();
-    final launchUri = Uri.parse('${_authorizationRequest.url}?$urlParams');
+    final urlCustomParams = _constructCustomUrlParams();
+    final launchUri =
+        Uri.parse('${_authorizationRequest.url}?$urlParams&$urlCustomParams');
```

Comment:
This will lead to an extra trailing & if there are no custom parameters. I would assume this wouldn't cause any problems, but have you tested this?

Reply:
I did think about that, and originally implemented a method that would prefix the ampersand to the custom parameters collection when building the url string. This would then output an empty string if no parameters were defined. I can confirm that a trailing ampersand does not cause any issues.

Reply:
I've though about this a bit more, I suggest adding the custom params in the function that constructs the urlParams. This means no trailing & and eliminates need for two functions for parameter construction.